opencog/truthvalue/AttentionValue.h

```diff
@@ -35,8 +35,10 @@ namespace opencog
  */
 
 //! stores attention in three components: short-term, long-term and very long-term
+//! This class is currently read-only, and assumed to be thread-safe. Beware of
+//! possible race conditions if it is changed
 class AttentionValue;
-typedef std::shared_ptr<AttentionValue> AttentionValuePtr;
+typedef std::shared_ptr<const AttentionValue> AttentionValuePtr;
```

Comment:
well, but that's wrong, AV's are not meant to be read-only, they are supposed to be mutable, and regularly changing, as the ECAN "agent" runs.